### PR TITLE
Fix inserting contracts in Navigator

### LINKED ISSUE #2271: Navigator Console 100.13.[12,13,14] does not register existing contracts
By running an application against a sandbox server 100.13.14 (same result with 12 and 13), contracts are created on the ledger. However, a subsequently started Navigator Console does not seem to see these contracts.

I managed to reproduce this error with the `quickstart-scala` application template, as follows:
1) create a new project from the template: 
```
$ daml new myname quickstart-scala
$ cd myname
```
2) build the dar file (and copy it to the expected place): 
```
$ daml build
$ ln -s .daml/dist dist # which works around a setup issue with this example
```
3) build and run the scala app
```
$ sbt 
sbt:quickstart-scala> application/runMain com.digitalasset.quickstart.iou.IouMain localhost 6865
... 
(which ends in a failure, but still creates some contracts)
```
4) start Navigator console with ` daml navigator console localhost 6865`. Then, after waiting for the actors to start, observe using `info` and `sql` that the `contract` table is empty, even though we have events that created contracts:

```
$ daml navigator console localhost 6865
   _  __          _           __
  / |/ /__ __  __(_)__ ____ _/ /____  ____
 /    / _ `/ |/ / / _ `/ _ `/ __/ _ \/ __/
/_/|_/\_,_/|___/_/\_, /\_,_/\__/\___/_/
                 /___/
Version 100.13.14
Welcome to the console. Type 'help' to see a list of commands.
Alice>info
   _  __          _           __
  / |/ /__ __  __(_)__ ____ _/ /____  ____
 /    / _ `/ |/ / / _ `/ _ `/ __/ _ \/ __/
/_/|_/\_,_/|___/_/\_, /\_,_/\__/\___/_/
                 /___/
Version 100.13.14 commit f70ccb2166d8d084ec6fc1efbde2f6b22dd37dc9
General info: 
...
Akka system:
  EUR_Bank: Actor running
  USD_Bank: Actor running
  Bob: Actor running
  Alice: Actor running
Local data:
  Alice:
    Packages: 3
    Contracts: 0
    Active contracts: 0
    Last transaction: 2
  Bob:
    Packages: 3
    Contracts: 0
    Active contracts: 0
    Last transaction: 2
...
Alice>sql select * from event  
... (shows create events)
Alice>sql select * from contract
<Empty>
Alice>event #2:1
Id: #2:1
ParentId: #2:0
TransactionId: 2
WorkflowId: Bob Workflow
Witnesses:
- Alice
Type: Created
Contract: #2:1
Template: Iou
Argument:
  issuer: Alice
  owner: Bob
  currency: USD
  amount: 1000.0
  observers:
  
ArgumentJson: {"observers":[],"issuer":"Alice","amount":"1000.0","currency":"USD","owner":"Bob"}
Alice>
```
5) Start a web navigator and verify that it can see the contract created above.
```
$ daml navigator server
```
browsing to `http://localhost:4000`, observe: 
image

### PR DESCRIPTION
**Pull request by rautenrieth-da**
Fixes #2271

### CODE REVIEW

**Comment by gerolf-da** [comment]
@rautenrieth-da: Could you add a line to `unreleased.rst` please?

**Comment by rautenrieth-da** [comment]
@gerolf-da I have added logging of all database errors. I don't think it's a proper solution to error handling, but it should make debugging errors much easier.

File: navigator/backend/src/main/scala/com/digitalasset/navigator/data/DatabaseActions.scala [hunk 1]
```diff
@@ -95,11 +96,19 @@ class DatabaseActions {
     } yield SqlQueryResult(colNames, data.map(_.map(Option(_).map(_.toString).getOrElse("null"))))
   }
 
-  def runQuery(query: String): Try[SqlQueryResult] = {
+  /** Returns the given value, logging failures */
+  private def logErrors[T](result: Try[T]): Try[T] =
+    result.recoverWith {
```
Comment (gerolf-da) [inline navigator/backend/src/main/scala/com/digitalasset/navigator/data/DatabaseActions.scala]: I think this is better expressed as:
```
result.failed.foreach { case NonFatal(t) => logger.error(....) }
result
```

The `recoverWith` indicates that some recovery checking is done, when it's actually not.

File: unreleased.rst [hunk 2]
```diff
@@ -22,3 +22,5 @@ HEAD — ongoing
   supported anymore. Use ``module_name`` and ``entity_name`` instead.
 - [Documentation] Improved the Maven pom.xml file for ``quickstart-java`` to better integrate with VS Code.
   See `#887 <https://github.com/digital-asset/daml/issues/887>`__.
+- [Sandbox] Fixed an issue when Navigator console did not see any contracts.
```
Comment (gerolf-da) [inline unreleased.rst]: Do we normally use the `Sandbox` category for `Navigator` changes?

**Comment by gerolf-da** [comment]
Thanks. That was my intention: making it easier to discover errors for now.

**Pull request merged**